extensions/2.0/Khronos/KHR_spz_gaussian_splats_compression/README.md

```diff
@@ -0,0 +1,144 @@
+
+At rest, the 3D Gaussian splats are stored within the SPZ compression format. Upon being decompressed, values are mapped to the glTF attributes defined by `KHR_gaussian_splatting` for rendering. The [conformance](#conformance) section defines what an implementation must do when encountering this extension, and how the extension interacts with the base specification.
+
+## Compressing 3D Gaussian splats using SPZ
+
+If a primitive contains an `extension` property which defines both `KHR_gaussian_splatting` and `KHR_spz_gaussian_splats_compression` then support for SPZ compression is required. There is no requirement for a backup uncompressed buffer.
+
+The extension must be listed in `extensionsUsed` alongside `KHR_gaussian_splatting`.
+
+```json
+  "extensionsUsed" : [
+    "KHR_gaussian_splatting",
+    "KHR_spz_gaussian_splats_compression"
+  ]
+```
+
+It must also be listed in `extensionsRequired`. When `KHR_spz_gaussian_splats_compression` is in use, the `KHR_gaussian_splatting` extension must also be listed as required.
+
+```json
+  "extensionsRequired" : [
+    "KHR_gaussian_splatting",
+    "KHR_spz_gaussian_splats_compression"
+  ]
+```
+
+### Inheritance from base extension
+
+As this extension extends the base extension, all components of the base extension are inherited. Any modifications to the base definition are explicitly defined in the sections below.
+
+### Schema Example
+
+Example SPZ extension shown below. This extension only affects any `primitive` nodes containting Gaussian splat data. Note that unlike the base `KHR_gaussian_splatting` extension, the `indices` property is excluded, and a `bufferView` is provided by the extension. This bufferview points to where the SPZ blob is stored.
+
+```json
+  "meshes": [{
+      "primitives": [{
+          "attributes": {
+            "POSITION": 0,
+            "COLOR_0": 1,
+            "_SCALE": 2,
+            "_ROTATION": 3,
+            "_SH_DEGREE_1_COEF_0": 4,
+            "_SH_DEGREE_1_COEF_1": 5,
+            "_SH_DEGREE_1_COEF_2": 6
+          },
+          "material": 0,
+          "mode": 0,
+          "extensions": {
+            "KHR_spz_gaussian_splats_compression": {
+              "bufferView": 0,
+            }
+          }
+        }]
+    }],
+  "buffers": [{
+      "uri": "0.bin",
+      "byteLength": 9753142
+    }],
+  "bufferViews": [{
+      "buffer": 0,
+      "byteLength": 9753142
+    }],
+```
+
+### Extension Properties
+
+#### bufferView
+
+This property points to the bufferView containing the Gaussian splat data compressed with SPZ.
+
+### Accessors
+
+Accessor requirements are modified from the base `KHR_gaussian_splatting` extension with the following adjustments to definition:
+
+ - SPZ compressed attributes must not include `bufferView` nor `byteOffset`. (See: [Conformance](#conformance))

```

Comment:
The `accessor.bufferView` property is defined as follows:
> The index of the buffer view. When undefined, the accessor **MUST** be initialized with zeros; `sparse` property or extensions **MAY** override zeros with actual values.

This enables graceful fallback behavior for engines that do not support SPZ.
- If SPZ is supported, accessor data is sourced from the decompressed SPZ blob
- If SPZ is not supported and `accessor.bufferView` is defined, accessor data is sourced as usual.
- If SPZ is not supported and `accessor.bufferView` is undefined, there may be other extensions providing the data.

Reply:
> If SPZ is supported, accessor data is sourced from the decompressed SPZ blob

Related to the comment at https://github.com/KhronosGroup/glTF/pull/2490#issuecomment-3097185006 :  Is there anything that should (or has to be) said about the `byteOffset` of these accessors, depending on whether or not the `bufferView` is present? 

(I think that the core(!) spec currently does not _disallow_ a `byteOffset`, even when _no_ `bufferView` is present. It's hard to imagine cases where this makes sense, but maybe there is no strong reason to _disallow_ it either...?)

Reply:
I see, https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#_accessor_byteoffset (I only looked in the table for some reason)

Reply:
I think graceful fallback is great, but I am struggling a bit to understand how I need to change this language to allow fallback. My understanding is that it's up to the runtimes to fill in the zeroes for an accessor without a `bufferView` property. 

Given that, I don't get the impression that this statement is in violation of that? The intent here is to indicate that an attribute that is compressed by SPZ can't specify a `bufferView` (or a `byteOffset`, but that is probably not needed to be mentioned here).

I can see some clarity problems with the original statement. Would it be sufficient to include your bullet points as clarification?

Reply:
What about attributes that aren't handled by SPZ? I'm primarily thinking of cases where someone may have a custom application attribute?

My gut is to specifically specify which attributes SPZ will provide data for, and then other attributes on a primitive are to be considered unhandled by SPZ. Is that alright?

Reply:
The SPZ extension spec should list supported attributes and how to get them from SPZ data. I think it's safe to assume (but should still be mentioned) that other attributes are either sourced as usual or handled by additional extensions.